linera-storage/src/lib.rs

```diff
@@ -128,6 +137,14 @@ pub trait Storage: Sized {
     /// Writes the given blob.
     async fn write_hashed_blob(&self, blob: &HashedBlob) -> Result<(), ViewError>;
 
+    /// Writes hashed certificates, hashed blobs and certificate
+    async fn write_hashed_certificate_values_hashed_blobs_certificate(
+        &self,
+        values: &[HashedCertificateValue],
+        blobs: &[HashedBlob],
+        certificate: &Certificate,
+    ) -> Result<(), ViewError>;
+
```

Comment:
Would it make sense to change the `Storage` trait so that instead of lots of very specialized methods like this one we have a notion of a higher-level "batch": something like a `struct StorageBatch` that contains a bunch of reads, writes and existence checks that will then all be executed at once?

Reply:
…or to extend the existing `Batch`?

Reply:
Yes, that would be my preference. Having the batch created externally and the write_back called externally. That would simplify the code.
About more complex data types, like `StorageBatch` why not. About changing the `Batch` I am against.

Reply:
What could be done is the following:
* Remove `write_certificates` and `write_certificate` from the code as they are not used.
* Remove `write_hashed_blob` and `write_hashed_blobs` as they are not used.
* Rename the `write_hashed_certificate_values_hashed_blobs_certificate`.

Of course, we could expose the `add_hashed_cert_value_to_batch`, `add_blob_to_batch`, `add_certificate_to_batch`, and `write_batch` since they are private functions now. However, I think that this would change the code significantly and I am not sure it would be a good idea to do that.